### PR TITLE
improve some output coloring

### PR DESCRIPTION
**Pull request by WebFreak001**
See #2347

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: source/dub/internal/logging.d [hunk 1]
```diff
@@ -380,3 +380,19 @@ string color(const string str, const Mode m = Mode.init)
...
+	tagWidthStack[tagWidthIndex] = tagWidth;
+	tagWidthIndex++;
+	assert(tagWidthIndex < tagWidthStack.length);
+	tagWidth = w;
+}
+
+void popTagWidth()
+{
+	tagWidth = tagWidthStack[--tagWidthIndex];
+}
```
Comment (Geod24) [inline source/dub/internal/logging.d:398]: Move it into its own struct please, instead of 3 globals

Comment (WebFreak001) [reply source/dub/internal/logging.d]: done

**Comment by Geod24** [comment]
I think the commits can be squashed BTW. The CI is erroring because of a minor difference in output, it seems

**Comment by WebFreak001** [comment]
squashed commits, fixed CI (issue was output format change caused the grep to be an invalid / non-matching regex)

**Review by Geod24 — APPROVED** [review]

**Pull request merged**